final_task/rss_reader/rss_parser.py

```diff
@@ -0,0 +1,155 @@
+
+    img_raw = re.search('><img src=".+"', data['image'])
+
+    if img_raw is not None:
+        img_clean = img_raw[0].replace('><img src="', '')
+        img_clean = re.sub('".+"', '', img_clean)
+        data['image'] = img_clean
+    else:
+        try:
+            img_try = rss.entries[limit].links[1]
+            data['image'] = img_try['href']
+        except IndexError:
+            data['image'] = None
+        except KeyError:
+            data['image'] = None
+
+    return data
+
+
+def print_rss(data: dict):
+    """This function prints data in a readable form"""
+
+    print('Feed: ' + data['feed'])
+    print('Title: ' + data['title'])
+    print('Date: ' + data['date'])
+    print('Link: ' + data['link'] + '\n')
+    print(data['description'], end='')
+
+    if data['image'] is not None:
+        print('Image: ' + data['image'])
+        print('. . . . . . . . . . . . . . . . . . . . . .')
+        print('')
+    else:
+        print('Image: image is not available.')
+        print('. . . . . . . . . . . . . . . . . . . . . .')
+        print('')
+
+
+def connect_rss(url: str) -> bool:
+    """This function tries to connect to RSS url
+    In case of failure, it reconnects in 10 seconds
+    """
+    for connection_tryouts in range(3):
+
+        try:
+            tryout = urllib.request.urlopen(url).getcode()
+            return tryout == 200
+
+        # Everything deals with time delay will be repeated
+        except urllib.error.HTTPError as http_err:
+            if http_err.code in (503, 504, 522, 524):
+                print('')
+                print('The server is not available:')
+                print('Trying to reconnect')
+
+                """We wait between connection tryouts"""
+                for time_delay in range(10):
+                    print('. ', end='')
+                    time.sleep(1)
+                print('')
+
+            # If server answer is a common code or something is not a common code
+            elif http_err.code in server_answer.keys():
+                print('The server can not be reached: Reason: %s' % server_answer[http_err.code])
+                return False
+            else:
+                print('Unknown error: %s' % http_err.code)
+                return False
+
+        # In case HTTPError is not working
+        except urllib.error.URLError as url_err:
+            print('The server can not be reached. Reason: %s' % url_err.reason)
+
+    return False
+
+
+def convert_date(date: str):
+    """This function converts date"""
+    month = {'Jan': '1',
+             'Feb': '2',
+             'Mar': '3',
+             'Apr': '4',
+             'May': '5',
+             'Jun': '6',
+             'Jul': '7',
+             'Aug': '8',
+             'Sep': '9',
+             'Oct': '10',
+             'Nov': '11',
+             'Dec': '12'}
+    day = date[5:7]
+    month_num = month[date[8:11]]
+    year = date[12:16]
```

Comment:
same: `datetime` lib

Reply:
Since the data from argsparse is a string, it is more convenient to use separate function instead of  datetime.

Reply:
This function is still redundant. It is better to use existing optimized solution instead of implementing new one